src/ceph-disk/ceph_disk/main.py

```diff
@@ -2123,20 +2179,101 @@ def __init__(self, args):
         super(PrepareBluestoreBlock, self).__init__(args)
 
     def get_space_size(self):
-        return 0  # get as much space as possible
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_size',
+        )
 
-    def desired_partition_number(self):
-        if self.args.block == self.args.data:
-            num = 2
+        if block_size is None:
+            return 0  # get as much space as possible
         else:
-            num = 0
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
+        num = 1
+        if getattr(self.args, "block") == getattr(self.args, "data"):
+            num += 1
+        if getattr(self.args, "block") == getattr(self.args, "block.db"):
+            num += 1
+        if getattr(self.args, "block") == getattr(self.args, "block.wal"):
+            num += 1
         return num
 
     @staticmethod
     def parser():
         return PrepareSpace.parser('block')
 
 
+class PrepareBluestoreBlockDB(PrepareSpace):
+
+    def __init__(self, args):
+        self.name = 'block.db'
+        super(PrepareBluestoreBlockDB, self).__init__(args)
+
+    def get_space_size(self):
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_db_size',
+        )
+
+        if block_size is None:
+            return 64 # MB, default value
+        else:
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
+        num = 1
+        if getattr(self.args, "block.db") == getattr(self.args, "data"):
+            num += 1
+        return num
+
+    @staticmethod
+    def parser():
+        parser = PrepareSpace.parser('block.db', positional = False)
```

Comment:
If I do this, I can't reuse the common part in ```PrepareSpace.parser```, for 3 other optional arguments, --%s-uuid, --%s-file, --%s-dev. ```PrepareBluestoreBlock, PrepareBluestoreBlockDB, PrepareBluestoreBlockWAL``` are reusing this part. 

What do you think?